.github/workflows/presubmit.yml

```diff
@@ -140,28 +59,14 @@ jobs:
       - uses: mpi4py/setup-mpi@v1
         with:
           mpi: intelmpi
-      - uses: actions/cache@v4
-        id: efficios_dep
-        env:
-          cache-name: cache-efficios_dep
-        with:
-          path: ~/efficios_dep/
-          key: ${{ runner.os }}-build-${{ env.cache-name }}
-      - run: sudo apt update; sudo apt install -y $APT_PACKAGE
-      - run: sudo gem install cast-to-yaml nokogiri babeltrace2 opencl_ruby_ffi metababel
-      - name: Load Babeltrace2
+      - uses: ./.github/actions/setup-thapi
+      - name: Make Install
         run: |
-          echo "$HOME/efficios_dep/bin" >> $GITHUB_PATH
-          echo "PKG_CONFIG_PATH=$HOME/efficios_dep/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
-          echo "CPATH=$HOME/efficios_dep/include:$CPATH" >> $GITHUB_ENV
-          echo "LD_LIBRARY_PATH=$HOME/efficios_dep/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
-          echo "LIBRARY_PATH=$HOME/efficios_dep/lib:$LIBRARY_PATH" >> $GITHUB_ENV
-      - run: mkdir -p build
-      - run: ./autogen.sh
-      - run: ../configure --prefix=`pwd`/ici
-        working-directory: build
-      - run: make -j install
-        working-directory: build
+          ./autogen.sh
+          mkdir -p build
+          cd build
+          ../configure --prefix=`pwd`/ici
+          make -j install
```

Comment:
I don't like these, this goes contradictory to the philosophy of github actions and increasing the diff for no good reason. I think that if we want to do that we should do so in a different PR.